src/libraries/Microsoft.PowerFx.Core/Localization/StringResources.cs

```diff
@@ -25,72 +24,193 @@ internal static class StringResources
         // If the dependency on ExternalStringResources is removed, this can be as well
         public static bool ShouldThrowIfMissing { get; set; } = true;
 
-        private static readonly ThreadSafeResouceManager _resourceManager = new ThreadSafeResouceManager();
+        internal static readonly IExternalStringResources LocalStringResources = new PowerFxStringResources("Microsoft.PowerFx.Core.strings.PowerFxResources", typeof(StringResources).Assembly, true);
 
-        [ThreadSafeProtectedByLockAttribute("_errorResources")]
-        private static readonly Dictionary<string, Dictionary<string, ErrorResource>> _errorResources = new Dictionary<string, Dictionary<string, ErrorResource>>(StringComparer.OrdinalIgnoreCase);
+        public static string Get(string resourceKey, string locale = null)
+        {
+            return TryGet(resourceKey, out string resourceValue, locale) ? resourceValue : null;
+        }
+
+        public static bool TryGet(string resourceKey, out string resourceValue, string locale = null)
+        {
+            bool success = LocalStringResources.TryGet(resourceKey, out resourceValue, locale);
+
+            if (success)
+            {
+                return true;
+            }
+
+            if (StringResources.ExternalStringResources != null && StringResources.ExternalStringResources.TryGet(resourceKey, out resourceValue, locale))
+            {
+                return true;
+            }
+
+            resourceValue = null;
+            return false;
+        }
 
         public static ErrorResource GetErrorResource(ErrorResourceKey resourceKey, string locale = null)
         {
             Contracts.CheckValue(resourceKey.Key, nameof(resourceKey));
             Contracts.CheckValueOrNull(locale, nameof(locale));
 
             // As foreign languages can lag behind en-US while being localized, if we can't find it then always look in the en-US locale
-            if (!TryGetErrorResource(resourceKey, out var resourceValue, locale))
+            if (resourceKey.ResourceManager.TryGetErrorResource(resourceKey, out ErrorResource resourceValue, locale))
             {
-                Debug.WriteLine(string.Format(CultureInfo.InvariantCulture, "ERROR error resource {0} not found", resourceKey));
-                if (ShouldThrowIfMissing)
-                {
-                    throw new System.IO.FileNotFoundException(resourceKey.Key);
-                }
+                return resourceValue;
             }
 
-            return resourceValue;
+            Debug.WriteLine(string.Format(CultureInfo.InvariantCulture, "ERROR error resource {0} not found", resourceKey));
+            throw new System.IO.FileNotFoundException(resourceKey.Key);
```

Comment:
>System.IO. [](http://example.com/codeflow?start=22&length=10)

[minor] add `using System.IO;`, since we don't use fully-qualified class names in other places